config/network-policies.yaml

```diff
@@ -0,0 +1,563 @@
+          cf-for-k8s.cloudfoundry.org/cf-system-ns: ""
+      podSelector:
+        matchLabels:
+          app.kubernetes.io/name: cf-api-server
+---
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: eirini
+  namespace: #@ data.values.system_namespace
+spec:
+  policyTypes:
+  - Ingress
+  - Egress
+  podSelector:
+    matchLabels:
+      name: eirini
+  ingress:
+  - from:
+    - podSelector:
+        matchLabels:
+          app.kubernetes.io/name: cf-api-deployment-updater
+      namespaceSelector:
+        matchLabels:
+          cf-for-k8s.cloudfoundry.org/cf-system-ns: ""
+    - podSelector:
+        matchLabels:
+          app.kubernetes.io/name: cf-api-clock
+      namespaceSelector:
+        matchLabels:
+          cf-for-k8s.cloudfoundry.org/cf-system-ns: ""
+    - podSelector:
+        matchLabels:
+          app.kubernetes.io/name: cf-api-worker
+      namespaceSelector:
+        matchLabels:
+          cf-for-k8s.cloudfoundry.org/cf-system-ns: ""
+    - podSelector:
+        matchLabels:
+          app.kubernetes.io/name: cf-api-server
+      namespaceSelector:
+        matchLabels:
+          cf-for-k8s.cloudfoundry.org/cf-system-ns: ""
+  egress:
+  - {}
+---
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: capi-kpack-watcher
+  namespace: #@ data.values.system_namespace
+spec:
+  policyTypes:
+  - Egress
+  podSelector:
+    matchLabels:
+      app.kubernetes.io/name: cf-api-kpack-watcher
+  egress:
+  - {}
+---
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: routecontroller
+  namespace: #@ data.values.system_namespace
+spec:
+  policyTypes:
+  - Egress
+  podSelector:
+    matchLabels:
+      app: routecontroller
+  egress:
+  - {}
+---
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: log-cache
+  namespace: #@ data.values.system_namespace
+spec:
+  policyTypes:
+  - Ingress
+  - Egress
+  podSelector:
+    matchLabels:
+      app: log-cache
+  ingress:
+  - from:
+    - podSelector:
+        matchLabels:
+          istio: ingressgateway
+      namespaceSelector:
+        matchLabels:
+          cf-for-k8s.cloudfoundry.org/istio-system-ns: ""
+    - podSelector:
+        matchLabels:
+          app.kubernetes.io/name: cf-api-server
+      namespaceSelector:
+        matchLabels:
+          cf-for-k8s.cloudfoundry.org/cf-system-ns: ""
+    - podSelector:
+        matchLabels:
+          app: fluentd
+      namespaceSelector:
+        matchLabels:
+          cf-for-k8s.cloudfoundry.org/cf-system-ns: ""
+  egress:
+  - to:
+    - podSelector:
+        matchLabels:
+          app.kubernetes.io/name: uaa
+      namespaceSelector:
+        matchLabels:
+          cf-for-k8s.cloudfoundry.org/cf-system-ns: ""
+    - podSelector:
+        matchLabels:
+          app.kubernetes.io/name: cf-api-server
+      namespaceSelector:
+        matchLabels:
+          cf-for-k8s.cloudfoundry.org/cf-system-ns: ""
+---
+kind: NetworkPolicy
```

Comment:
You could give a permissive policy to the `prometheus` -- allow it to egress to anything -- and then for the components that have metrics endpoints they could either:

1. Write an ingress policy to explicitly allow prometheus (this would end up in a lot of entries probably) for each component
2. We could coordinate and have system components serve their metrics scrape endpoints on a specific well-known port and have a blanket allow ingress rule for that port
3. Assuming you label prometheus with a well-known label, have a blanket allow ingress rule for the namespace that says that pods should allow ingress from something with that label
3. Some mixture of the above?

Reply:
Options 2/3 seem like good options with 3 probably being the most ideal. Duplicating ingress across each component would be very unfortunate.

Reply:
What would Option 3 policy look like?

Reply:
I like option 3 as well. Regarding 2, I think CAPI needed to use 9102 for statsd_exporter.

I really like the loop idea and feel like could make implementing a netpol for prometheus simple if we are leaning towards option 1. I might even vote to put prometheus in its own namespace since it doesn't seem like we're gaining anything by putting it in cf-system anymore.

Reply:
Option 3 would look similar to this policy that allows anything in `istio-system` to talk to all of the Pods in `cf-system`:
https://github.com/cloudfoundry/cf-for-k8s/pull/270/files#diff-652a3e228db9cbb604d4ba5801be220fR47

A similar policy could be made that matches to `prometheus`.

Reply:
> I might even vote to put prometheus in its own namespace since it doesn't seem like we're gaining anything by putting it in cf-system anymore.

I think `cf-for-k8s` as a project should have a larger discussion on this. I think there are some operational benefits to consolidating "cf related stuff" in `cf-system`, but am curious to hear others' thoughts here and would love project-wide guidance on the subject. 😅

cc @jenspinney @jamespollard8 @davewalter

Reply:
I'm inclined to lean into "do what works" here (as opposed to following rules about consistently consolidating or separating components into namespaces).

So, maybe that means put things in cf-system by default AND have no qualms about putting prometheus in its own namespace if that helps us operationally/maintenance-wise.

I'm definitely interested in hearing from others (eg Jen, Dave, etc)

Reply:
> Option 3 would look similar to this policy that allows anything in `istio-system` to talk to all of the Pods in `cf-system`:
> https://github.com/cloudfoundry/cf-for-k8s/pull/270/files#diff-652a3e228db9cbb604d4ba5801be220fR47
> 
> A similar policy could be made that matches to `prometheus`.

I see, and we'd need one of these for all namespaces then. 

I'd also like guidance about what namespaces we should be using as well.

Reply:
Is this still an open question? If not, could someone please update this thread and resolve it?

Reply:
I think the docs we wrote around [adding a NetworkPolicy](https://github.com/cloudfoundry/cf-for-k8s/blob/7a9ee9571538c01534661291734c6f510647ff5a/docs/component_authors/networking.md#writing-network-policies) for new components addresses this.